### PR TITLE
feat(FX-4305): add ability to hide bottom tabs on specific screen

### PR DESCRIPTION
**Pull request by dimatretyak**
This PR resolves [FX-4305] 

#### iOS
https://user-images.githubusercontent.com/3513494/205075962-6e138520-1512-43ee-b33b-6ce760de1e22.mp4

#### Android
https://user-images.githubusercontent.com/3513494/205077788-d45d93b6-cf39-4449-b8df-f309818a7c95.mp4

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- add ability to hide bottom tabs on specific screen - dimatretyak

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[FX-4305]: https://artsyproduct.atlassian.net/browse/FX-4305?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/BottomTabs/BottomTabs.tsx [hunk 1]
```diff
@@ -26,8 +28,13 @@ export const BottomTabs: React.FC = () => {
...
 
   const { bottom } = useScreenDimensions().safeAreaInsets
+
+  if ((focusedRoute?.params as any)?.shouldHideBottomTab) {
+    return <></>
+  }
+
   return (
-    <Flex style={{ paddingBottom: bottom }}>
+    <Flex position="absolute" left={0} right={0} bottom={0} paddingBottom={bottom} bg="white100">
```
Comment (dimatretyak) [inline src/app/Scenes/BottomTabs/BottomTabs.tsx:37]: `position: absolute"` is used here to solve the problem with displaying gray empty space (see attached video). 

https://user-images.githubusercontent.com/3513494/205076919-dca6a5d3-730d-4558-8b7d-5a8f1a311e97.mp4

File: src/app/Scenes/Artwork/Artwork.tsx [hunk 2]
```diff
@@ -151,6 +153,10 @@ export const Artwork: React.FC<ArtworkProps> = ({
     }
   }, [])
 
+  useLayoutEffect(() => {
+    setBottomTabVisibilityForCurrentScreen(false)
+  }, [])
```
Comment (gkartalis) [inline src/app/Scenes/Artwork/Artwork.tsx:158]: 🔥👌🥇

File: src/app/Scenes/BottomTabs/BottomTabs.tsx [hunk 3]
```diff
@@ -26,8 +28,13 @@ export const BottomTabs: React.FC = () => {
   const enableMyCollectionInsights = useFeatureFlag("AREnableMyCollectionInsights")
 
   const { bottom } = useScreenDimensions().safeAreaInsets
+
+  if ((focusedRoute?.params as any)?.shouldHideBottomTab) {
+    return <></>
```
Comment (gkartalis) [inline src/app/Scenes/BottomTabs/BottomTabs.tsx:33]: **question:** is this intentional? Why not `null`

File: src/app/Scenes/BottomTabs/BottomTabs.tsx (hunk 1 above)
Comment (gkartalis) [reply src/app/Scenes/BottomTabs/BottomTabs.tsx:37]: 👏

**Review by gkartalis — APPROVED** [review]
🥇 Can't thank you enough for this fix! 🙏

File: src/app/Scenes/BottomTabs/BottomTabs.tsx (hunk 3 above)
Comment (dimatretyak) [reply src/app/Scenes/BottomTabs/BottomTabs.tsx:33]: nope, we can use `null` also. i'll change it

**Comment by gkartalis** [comment]
also tested on my device and the previous issue is not reproducible any more!

File: src/app/Scenes/BottomTabs/BottomTabs.tsx (hunk 3 above)
Comment (dimatretyak) [reply src/app/Scenes/BottomTabs/BottomTabs.tsx:33]: done

File: src/app/Scenes/BottomTabs/BottomTabs.tsx (hunk 1 above)
Comment (pvinis) [reply src/app/Scenes/BottomTabs/BottomTabs.tsx:37]: I don't get this. why is the grey bit showing?

File: src/app/Scenes/Artwork/Artwork.tsx (hunk 2 above)
Comment (pvinis) [reply src/app/Scenes/Artwork/Artwork.tsx:159]: hm. could we have this in the routes? the boolean for hiding the tabbar?

File: src/app/Scenes/BottomTabs/setBottomTabVisibilityForCurrentScreen.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { __unsafe_mainModalStackRef } from "app/NativeModules/ARScreenPresenterModule"
```
Comment (pvinis) [inline src/app/Scenes/BottomTabs/setBottomTabVisibilityForCurrentScreen.ts:1]: wow this was a thing already?

**Review by pvinis — CHANGES_REQUESTED** [review]
I don't want to block this, but I do want to understand the grey bit and see if we could move from the hook to the routes before this is merged 🙏

File: src/app/Scenes/BottomTabs/BottomTabs.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/app/Scenes/BottomTabs/BottomTabs.tsx:37]: This is an iOS-specific bug that does not reproduce in all cases/not always

File: src/app/Scenes/BottomTabs/setBottomTabVisibilityForCurrentScreen.ts (hunk 4 above)
Comment (dimatretyak) [reply src/app/Scenes/BottomTabs/setBottomTabVisibilityForCurrentScreen.ts:1]: yep

**Comment by dimatretyak** [comment]
> if we could move from the hook to the routes

I don't quite understand what you mean. if it's not difficult for you, can you please describe it in a little more detail?

**Comment by dimatretyak** [comment]
> I don't want to block this, but I do want to understand the grey bit

[Here](https://github.com/react-navigation/react-navigation/issues/10432) you can find a little more context about the problem and [some solutions](https://github.com/react-navigation/react-navigation/issues/10432#issuecomment-1298324534) (only the first proposed solution solved the problem in our case)

File: src/app/Scenes/Artwork/Artwork.tsx (hunk 2 above)
Comment (dimatretyak) [reply src/app/Scenes/Artwork/Artwork.tsx:159]: the main problem is that we have to dynamically hide the bottom tab bar depending on the some condition (in our case, the bottom tab bar will be hidden if the `isForSale` of artworks is set to `true` and sticky bottom content will be displayed. [Jira ticket](https://artsyproduct.atlassian.net/browse/FX-4285) for it)

**Review by pvinis — APPROVED** [review]
ah. so its not a problem we created, its a lib thing. cool cool, thanks for the links!

**Pull request merged**